dev/tools/lib/codesign.dart

```diff
@@ -0,0 +1,360 @@
+        platform = checkouts.platform,
+        stdio = checkouts.stdio,
+        processManager = checkouts.processManager {
+    argParser.addFlag(
+      kVerify,
+      help:
+          'Only verify expected binaries exist and are codesigned with entitlements.',
+    );
+    argParser.addFlag(
+      kSignatures,
+      defaultsTo: true,
+      help:
+          'When off, this command will only verify the existence of binaries, and not their\n'
+          'signatures or entitlements. Must be used with --verify flag.',
+    );
+    argParser.addOption(
+      kUpstream,
+      defaultsTo: FrameworkRepository.defaultUpstream,
+      help: 'The git remote URL to use as the Flutter framework\'s upstream.',
+    );
+    argParser.addOption(
+      kRevision,
+      help: 'The Flutter framework revision to use.',
+    );
+  }
+
+  final Checkouts checkouts;
+  final FileSystem fileSystem;
+  final Platform platform;
+  final ProcessManager processManager;
+  final Stdio stdio;
+
+  FrameworkRepository _framework;
+  FrameworkRepository get framework => _framework ??= FrameworkRepository.hostRepoAsUpstream(checkouts);
+
+  @visibleForTesting
+  set framework(FrameworkRepository framework) => _framework = framework;
+
+  @override
+  String get name => 'codesign';
+
+  @override
+  String get description =>
+      'For codesigning and verifying the signatures of engine binaries.';
+
+  @override
+  void run() {
+    if (!platform.isMacOS) {
+      throw ConductorException(
+          'Error! Expected operating system "macos", actual operating system is: '
+          '"${platform.operatingSystem}"');
+    }
+
+    if (argResults['verify'] as bool != true) {
+      throw ConductorException(
+          'Sorry, but codesigning is not implemented yet. Please pass the '
+          '--$kVerify flag to verify signatures.');
+    }
+
+    String revision;
+    if (argResults.wasParsed(kRevision)) {
+      stdio.printError('Warning! When providing an arbitrary revision, the contents of the cache may not');
+      stdio.printError('match the expected binaries in the conductor tool. It is preferred to check out');
+      stdio.printError('the desired revision and run that version of the conductor.\n');
+      revision = argResults[kRevision] as String;
+    } else {
+      revision = (processManager.runSync(
+        <String>['git', 'rev-parse', 'HEAD'],
+        workingDirectory: checkouts.directory.path,
+      ).stdout as String).trim();
+    }
+
+    framework.checkout(revision);
+
+    // Ensure artifacts present
+    framework.runFlutter(<String>['precache', '--ios', '--macos']);
+
+    verifyExist();
+    if (argResults[kSignatures] as bool) {
+      verifySignatures();
+    }
+  }
+
+  /// Binaries that are expected to be codesigned and have entitlements.
+  ///
+  /// This list should be kept in sync with the actual contents of Flutter's
+  /// cache.
+  List<String> get binariesWithEntitlements {
+    return <String>[
```

Comment:
Can you sort this list to make it easier to read?

Reply:
good idea

Reply:
done